[NETBEANS-54] Module Review

bondolo opened the pull request:
Switches HtmlElements.java to use ASF license.

Fixes #67

dschoorl committed:
[NETBEANS-54] Module Review api.annotations.common

- external library: jsr305 locense file updated with missing header 
  fields (see http://wiki.netbeans.org/DevFaqExternalLibraries), so that
  'ant verify-libs-and-licenses' no longer complains about jsr305. Maven 
  coordinates added, with side-effect that artifact name changed to
  jsr305-2.0.0.jar. Notice added for jcip-annotations. 
  NOTE: Checksum in binaries-list for jsr305 changed (artifact build
  with different JRE version).

- checked Rat report; changed license header in apichanges.xml, removed
  Oracle license header in file under META-INF/services (excluded from
  Rat)

- skimmed through the module, did not spot files that were not Oracle's
  to donate.

matthiasblaesing committed:
[NETBEANS-54] Module review xml.text.obsolete90

- no external library
- checked Rat report; added license header to testfiles and ensured 
  tests still run cleanly, removed unused file, that only differs from
  neighbouring file by license
- skimmed through the module, did not notice additional problems

matthiasblaesing committed:
[NETBEANS-54] Module review xml.core

- no external library
- checked Rat report; added license header to testfiles
- skimmed through the module, did not notice additional problems

matthiasblaesing committed:
[NETBEANS-54] Module review xml.retriever

- no external library
- checked Rat report; added license header to testfiles
- skimmed through the module, did not notice additional problems

matthiasblaesing committed:
[NETBEANS-54] Module review xml.multiview

- no external library
- checked Rat report; added license header to testfiles, manually 
  updated unconverted license headers
- skimmed through the module, did not notice additional problems

matthiasblaesing committed:
[NETBEANS-54] Module review xsl

- no external library
- checked Rat report; added testfiles to exclude list for rat
- skimmed through the module, did not notice additional problems

matthiasblaesing committed:
[NETBEANS-54] Module review xml.text

- no external library
- checked Rat report; added license to many test files and adjusted
  tests to match these; exclude GUI visible sample data and special
  test data
- skimmed through the module, did not notice additional problems

matthiasblaesing committed:
[NETBEANS-54] Modules review xml.axi

- no external library
- checked Rat report: excluded test-data from rat (no IP value) 
- disable tests that fail because referenced files were not donated
- skimmed through module, did not notice additional problems

matthiasblaesing committed:
[NETBEANS-54] Modules review xml.xam

- no external library
- checked Rat report: manually updated missed license header and added
  missing headers
- skimmed through module, did not notice additional problems

matthiasblaesing committed:
[NETBEANS-54] Modules review xml.schema.completion
    
- no external library
- checked Rat report: excluded test-data from rat (no IP value), updated
  XMLSchema.xsd and added w3c document and software license and
  made sure license is picked up by rat report
- disable tests that fail because referenced files were not donated
- skimmed through module, did not notice additional problems

matthiasblaesing committed:
[NETBEANS-54] Modules review xml.tax
    
- no external library
- checked Rat report: added missing headers, ignored i10n.list
- skimmed through module, did not notice additional problems

matthiasblaesing committed:
[NETBEANS-54] Modules review xml.jaxb.api

- external library jsr173_api.jar was removed, as StAX api was moved
  into java platform
- external libraries jaxb-api and activation were moved to maven
  coordinates and version was corrected
- checked Rat report: updates license header in Bundle.properties
- skimmed through module, did not notice additional problems

matthiasblaesing committed:
[NETBEANS-54] Modules review xml.xdm

- no external library
- checked Rat report: excluded test-data from rat (no IP value)
- disable tests that fail because referenced files were not donated
- skimmed through module, did not notice additional problems

matthiasblaesing committed:
[NETBEANS-54] Modules review xml.catalog

- removed external binary (contained XML Schemas and DTDs), added 
  license header to files and added them to the source code, where the
  binary was expanded at build time
- checked Rat report: added missing headers
- skimmed through module, did not notice additional problems

matthiasblaesing committed:
[NETBEANS-54] Modules review xml.lexer

- no external library
- checked Rat report: excluded test-data from rat (no IP value)
- disable tests that fail because referenced files were not donated
- skimmed through module, did not notice additional problems

matthiasblaesing committed:
[NETBEANS-54] Modules review xml.tools

- no external library
- checked Rat report: excluded test-data from rat (no IP value)
- skimmed through module, did not notice additional problems

matthiasblaesing committed:
[NETBEANS-54] Modules review xml.schema.model

- no external library
- checked Rat report: excluded test-data from rat (no IP value), covered
  zip files contains simple XSD samples as groups
- disable tests that fail because referenced files were not donated
- skimmed through module, did not notice additional problems

matthiasblaesing committed:
[NETBEANS-54] Modules review xml.wsdl.model

- no external library
- checked Rat report: manully updated missed license headers
- readded undonated wsdl-2004-08-24.xsd from source and added license
  to it (fixes functionality)
- disable tests that fail because referenced files were not donated
- skimmed through module, did not notice additional problems

matthiasblaesing committed:
[NETBEANS-54] Modules review xml

- no external library
- checked Rat report: excluded test-data from rat (no IP value),
  updated/added license headers, exclude templates from rat (jira
  issue was raised)
- disable tests that fail because referenced files were not donated
- skimmed through module, did not notice additional problems

jlahoda committed:
[NETBEANS-54] Module Review applemenu

-using an external library instead of inline com/apple/eawt classes
-saner handling of "external" java.awt classes (not unpacking them into the source directory)

vieiro committed:
[NETBEANS-54] Module Review libs.lucene

  - Added lucene notice file from https://github.com/apache/lucene-solr/blob/releases/lucene-solr/3.5.0/lucene/NOTICE.txt
  - Added the maven coordinates.
  - No other issues found.

vieiro committed:
[NETBEANS-54] Module Review libs.xerces

  - Updated maven coordinates for Xerces 2.8.0 (Apache 2).
  - Added xerces notice from https://svn.apache.org/repos/asf/xerces/java/tags/Xerces-J_2_8_0/NOTICE
  - Updated nbproject/project.properties as requested.

vieiro committed:
[NETBEANS-54] Module Review libs.jvyamlb

  - Version 0.2.7 is not available in maven.central
  - Downgraded to 0.2.6 to see how it behaves in travis build
  - See issue #82 https://issues.apache.org/jira/browse/NETBEANS-82

vieiro committed:
[NETBEANS-54] Module Review libs.smack

  - Updated maven coordinates.
  - Updated nbproject/project.properties and nbproject/project.xml with the maven names.
  - Passing tests. No license headers missing.
  - Still awaiting for travis beers.

Commit:
[NETBEANS-54] Module Review

Switches HtmlElements.java to use ASF license.

emilianbold commented on html.lexer/src/org/netbeans/lib/html/lexer/HtmlElements.java:
```diff
@@ -1,46 +1,22 @@
+ * specific language governing permissions and limitations
+ * under the License.
  */
+
```
This extra new line is not necessary.

emilianbold reviewed (changes requested):
You must also exclude in nbbuild/build.xml ("rat" task) the test data from html.lexer/test/unit/data/testfiles/**

bsungur committed:
[NETBEANS-54] Module Review languages.diff

Exclude DiffExample.diff from Rat since a license header would be visible to users in Fonts and Color settings.

Commit:
[NETBEANS-54] Updated binaries to use maven lookups - 
Had to change from using a jar called:
org.eclipse.jgit-3.6.2.201501210735-r-nosignature.jar to
org.eclipse.jgit-3.6.2.201501210735-r.jar
Since the previous one wasnt in maven.  NO NOTICE file added - Not sure it needs one since its under EDL - see https://github.com/eclipse/jgit/tree/v3.6.2.201501210735-r

Commit:
[NETBEANS-54] Updated binaries to use maven lookups - 
Had to change from using a jar called:
org.eclipse.jgit.java7-3.6.2.201501210735-r-nosignature.jar to
org.eclipse.jgit.java7-3.6.2.201501210735-r.jar
Since the previous one wasn't in maven.  NO NOTICE file added - Not sure it needs one since its under EDL - see https://github.com/eclipse/jgit/tree/v3.6.2.201501210735-r

Commit:
[NETBEANS-54] Module Review junitlib

Commit:
[NETBEANS-54] Module Review java.completion

trevlewis committed:
[NETBEANS-54] Module Review selenium2.java

tomazst committed:
[NETBEANS-54] Module Review projectui

Exclude license files from Rat.

trevlewis committed:
[NETBEANS-54] Module Review spellchecker.dictionary_en

Commit:
[NETBEANS-54] Updated rat exclusions to include hudson.subversion/test/unit/src/org/netbeans/modules/hudson/subversion/sample-entries-file since this file is test data. - including a license will break tests.

neilcsmith-net committed:
[NETBEANS-54] Add .palette_item files to Rat exclude.

neilcsmith-net committed:
[NETBEANS-54] Replace Oracle licenses with Apache on Form template files.

Commit:
[NETBEANS-54] Added missing license and included 2 exclusion rules for test data

Commit:
[NETBEANS-54] Added missing licenses and replaced oracle license in a single file.

AnEmortalKid committed:
[NETBEANS-54] Module Review debugger.jpda

Exclude JspLineBreakpointApp from Rat task since this file cannot have a header.

nwholloway committed:
[NETBEANS-54] Module Review java.editor

- No external library
- Reviewed RAT report
-- Excluded Java template file
-- Updated license in a test configuration file
-- Excluded test data files

nwholloway committed:
[NETBEANS-54] Module Review java.editor.base

No external library
Reviewed RAT report
- Excluded test data files

vieiro committed:
[NETBEANS-54] Module Review mercurial

  - Added license to l10n.list
  - Added templates & test files to rat exclusion list
  - No other licensing issues found.

vieiro committed:
[NETBEANS-54] Module Review libs.cglib

  - Updated maven coordinates for the external binary.
  - Added cglib-2.2 license from https://github.com/cglib/cglib
  - No other licensing issues found.

vieiro committed:
[NETBEANS-54] Module Review dlight.nativeexecution

  - The external/exechlp-1.0.zip binary is generated by the scripts and code in tools/
  - So I'm assuming this is an Apache 2.0 license as well.
  - Added license headers to makefiles and different scripts
  - No other licensing headers required.
  - Test org.netbeans.modules.nativeexecution.api.util.ConnectionManagerTest failed in my box.

junichi11 committed:
[NETBEANS-54] Module Review refactoring.api

- no external library
- checked Rat report: update the license header in translate.names
- skimmed through the module, did not notice additional problems

vieiro committed:
[NETBEANS-54] Module Review extexecution.process

  - Updated coordinates for libpam4j from Maven central (MIT)
  - winp binary (MIT) in this directory is a patched version from the one in Maven. Added maven coordinates as a comment in binaries-list.
  - libpam4j (MIT) is not available in Maven central, the homepage was hosted in kenai, now gone.
  - Unit tests passing.
  - No other licensing issues found.

vieiro committed:
[NETBEANS-54] Module Review ide.kit

  - No external binaries.
  - Added Apache license to different xml/properties files and release/VERSION.txt.
  - Added rat exclusion list for all qa-functional/**/*.txt files.
  - Remaining files to be handled centrally.
  - No other licensing issues found.

Commit:
[NETBEANS-54] Module Review java.editor create shell script

Commit:
[NETBEANS-54] Module Review nbjunit

ebarboni committed:
[PATCH] Produce junit report from Apache Rat and clusters.properties

vieiro committed:
[NETBEANS-54] Module Review spi.debugger.ui

  - No external binaries.
  - Removed previous license from many META-INF files.
  - Two files to be handled centrally (*.sig, manifest.mf)
  - Two tests failing (both before and after changes): org.netbeans.api.debugger.ProvidersAnnotationTest and org.netbeans.api.debugger.WatchesTest
  - No other licensing issues found.

vieiro committed:
[NETBEANS-54] Module Review o.apache.ws.commons.util

  - Updated maven coordinates for Apache WS Commons Util (Apache 2.0)
  - No notice found at http://svn.apache.org/viewvc/webservices/commons/trunk/modules/util/
  - No other licensing issues found (manifest.mf to be handled centrally)

ebarboni committed:
Use InputSource to parse xml avoid encoding issue

matthiasblaesing committed:
[NETBEANS-54] Add ALv2 header to l10n.list files

This closes: #121

matthiasblaesing committed:
[NETBEANS-54] Recursively exclude generated project files

matthiasblaesing committed:
[NETBEANS-54] Added/updated ALv2 header for project.properties

matthiasblaesing committed:
[NETBEANS-54] Added/updated ALv2 header for build.xml

matthiasblaesing committed:
[NETBEANS-54] Added/updated ALv2 header for project.xml

matthiasblaesing committed:
[NETBEANS-54] Add license to platform.properties

vieiro committed:
[NETBEANS-54] Module Review o.apache.xmlrpc

  - Added three entries in binaries-list with Maven coordinates for Apache XMLRPC 3.0 files (Apache 2.0)
  - No notice files found at https://svn.apache.org/repos/asf/webservices/archive/xmlrpc/tags/XMLRPC_3_0/
  - No other licensing issues found.

vieiro committed:
[NETBEANS-54] Module Review o.eclipse.core.contenttype

  - Added maven coordinates for for org/eclipse/core/contenttype (EPL 1.0)
  - Updated nbproject/* to reflect the change in the binary.
  - NOTE: Maven jar file is signed (original was not).
  - No other licensing issues found.

matthiasblaesing committed:
[NETBEANS-54] Added/updated ALv2 header in META-INF/services/*

matthiasblaesing committed:
[NETBEANS-54] Module Review junitlib

- Remove license file for non-existing binary
- Fix license file for junit4-12
- Move notice file to common name definition

vieiro committed:
[NETBEANS-54] Module Review o.eclipse.core.jobs

  - Exact version 3.5.101 could not be found in Maven, downgrading to 3.5.100
  - Updated binaries-list with maven coordinates for eclipse core jobs 3.5.100 (EPL 1.0)
  - Updated nbproject/* accordingly.
  - No other licensing issues found.

matthiasblaesing committed:
[NETBEANS-54] Module Review java.completion

 - Fix unittests after changed file structure caused by added license

tomazst committed:
[NETBEANS-54] Module Review css.prep

tuyenngoctran committed:
[NETBEANS-54] Module Review junit.ui

mirkoperillo committed:
[NETBEANS-54] Module Review java.api.common

Excluded CommandLine.tokens from Rat

matthiasblaesing committed:
[NETBEANS-54] Module Review api.annotations.common

- Rename jcip-notice.txt -> jsr305-2.0.0-license.txt
  to be in line with the name of the corresponding license file

bondolo committed:
[NETBEANS-54] Module Review

Switches HtmlElements.java to use ASF license.

bondolo committed:
[NETBEANS-54] Module Review

Switches HtmlElements.java to use ASF license.

bondolo commented:
Should automatically resolve when you pull from apache repo

emilianbold reviewed (changes requested):
I cannot review this since you did a bad rebase or something and I see 655 files changed.

A lot of commits appear as 'committed with Mike Duigou' which is a sign of rebase to me.

Please do a proper merge / rebase and push the new changes.

emilianbold commented:
Closing this since you opened https://github.com/emilianbold/incubator-netbeans/pull/327
